api/types/src/convert.rs

```diff
@@ -112,6 +120,10 @@ impl<'a, R: MoveResolver + ?Sized> MoveConverter<'a, R> {
             success: info.status().is_success(),
             vm_status: self.explain_vm_status(info.status()),
             accumulator_root_hash: accumulator_root_hash.into(),
+            changes: write_set
+                .into_iter()
+                .filter_map(|(sk, wo)| self.try_into_write_set_change(sk, wo).ok())
```

Comment:
1. this interprets the code / resource at the version of the converter, not `version` passed in, which is meaningless. 
2. I suspect the performance hit will be devastating.

Reply:
Oh I overstated, sorry.
1. resource value is interpreted by the type definition at the version of the converter, not the `version` passed in -- can be a issue if a module is updated.

Reply:
Is this an issue for the write set alone, or is this something that would also affect all resource serialization?

Reply:
we're not planning on allowing module updates for the time being: given always pulling info from the same version as a tx would be a large change, would you be okay with a `TODO`?